### PR TITLE
[s] Fixes altclick exploits

### LINKED ISSUE #35450: Ghosts can draw knives/items from boots
Bagil
Round # 83272

Title, ghosts can pull knives and other items by alt-clicking boots that have contents.

To replicate just observe  anyone with boots and are likely to have something stored in them like shaft-miners and alt-click their boots. If there's anything in them it's now yours, and you can freely fly it around anywhere (Brig is a good start) and drop it off using the drop hotkey (q).

Should probably be fixed because 2spooky, can also have a significant round impact if you stole something 'really' useful like a traitor's energy pen or similar.

### LINKED ISSUE #35498: You can unload pet carriers any distance away
[Reproduction]: Unload a pet carrier like normal but unload it half the station away

[For Admins]: # (Oddities induced by var-edits and other admin tools are not necessarily bugs. Verify that your issues occur under regular circumstances before reporting them.)

### PR DESCRIPTION
**Pull request by ShizCalev**
fixes #35450
fixes #35498

Missing adjacency checks, stat checks, ect. This allowed players to teleport items around, open/close things without being near them, ect.

:cl: ShizCalev
fix: Corrected a number of missing checks when using alt-click actions. Please report any strange behavior to a coder.
/:cl:

### CODE REVIEW

**Comment by Cyberboss** [comment]
@tgstation/commit-access We honestly need a layer above `AltClick`. It's too much work for every proc to remember everything it needs to prep the entire click chain. I'm thinking an `/atom/proc/AlternateAction()` which will be called by the click chain similar to `attack_hand()`. Thoughts?

**Comment by AnturK** [comment]
Yep.

**Comment by AnturK** [comment]
Besides the need for the wrapper, this makes pAI's able to do all these actions from any place at any time. And probably more inconsistencies, because canUseTopic is pretty mediocre approximation for what's needed in these checks, i don't really like using it as the default interaction check.

File: code/modules/mob/living/silicon/pai/pai.dm [hunk 1]
```diff
@@ -185,6 +185,8 @@
 // See software.dm for Topic()
 
 /mob/living/silicon/pai/canUseTopic(atom/movable/M, be_close=FALSE, no_dextery=FALSE)
+	if(be_close && !in_range(M, src))
```
Comment (ShizCalev) [inline code/modules/mob/living/silicon/pai/pai.dm]: @AnturK Already fixed that.

File: code/game/machinery/hologram.dm [hunk 2]
```diff
@@ -137,10 +137,6 @@ Possible to do for anyone motivated enough:
 
 	return ..()
 
-/obj/machinery/holopad/AltClick(mob/living/carbon/human/user)
-	if(isAI(user))
```
Comment (ShizCalev) [inline code/game/machinery/hologram.dm]: if mob/living/carbon/human/user isAI. nice meme

This should've been handled by AIAltClick anyway.

**Pull request merged**